### PR TITLE
Fix msr package

### PR DESCRIPTION
**Pull request by GanShun**
The msr package was erroring in an init due to my mistake. Removed the
init and just added a helper function for all CPUs.

Signed-off-by: Gan Shun Lim <ganshun@gmail.com>

Sorry about that! can you try this out and see if its better?

### CODE REVIEW

File: pkg/msr/msr_linux.go [hunk 1]
```diff
@@ -17,32 +16,27 @@ import (
...
-		log.Print(errs)
+func AllCPUs() (CPUs, error) {
+	c, errs := GlobCPUs("*")
+	if errs != nil || len(c) == 0 {
+		return nil, fmt.Errorf("error finding all cpus, maybe try modprobe msr? : %v", errs)
 	}
+	return c, nil
 }
 
 // GlobCPUs allow the user to specify CPUs using a glob as one would in /dev/cpu
```
Comment (rminnich) [inline pkg/msr/msr_linux.go]: Why not just Glob
msr.Glob()

?

File: pkg/msr/msr_linux.go [hunk 2]
```diff
@@ -17,32 +16,27 @@ import (
...
-var AllCPUs CPUs
-
-func init() {
-	var errs []error
-
-	if AllCPUs, errs = getAllCPUs(); errs != nil {
-		// Should I fatal here? panic?
-		log.Print("Failed to find all CPUs from /dev/cpu")
-		log.Print(errs)
+func AllCPUs() (CPUs, error) {
```
Comment (rminnich) [inline pkg/msr/msr_linux.go]: All
then you have 
msr.All()

File: cmds/core/lockmsrs/lockmsrs.go [hunk 3]
```diff
@@ -79,7 +79,10 @@ func main() {
 		log.Fatalf("Only Intel CPUs supported, expected GenuineIntel, got %v", cpuid.VendorIdentificatorString)
 	}
 
-	cpus := msr.AllCPUs
+	cpus, err := msr.AllCPUs()
```
Comment (rminnich) [inline cmds/core/lockmsrs/lockmsrs.go]: msrs, err := msr.All

so it does not stutter?

Comment (hugelgupf) [reply cmds/core/lockmsrs/lockmsrs.go]: I don't agree. It's not immediately obvious from msr.All or msr.Glob what you're getting back

File: pkg/msr/msr_linux.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/msr/msr_linux.go]: Glob what? All MSR values possible on my platform? Or all CPUs? Or all whatevers? Not obvious.

File: cmds/core/lockmsrs/lockmsrs.go (hunk 3 above)
Comment (GanShun) [reply cmds/core/lockmsrs/lockmsrs.go]: Yeah I agree with chris here, msr.All sounds like all msrs.

**Comment by GanShun** [comment]
I'm going to submit this first in case other people run into the same issues, but @JulienVdG please lmk if it works for you

**Comment by JulienVdG** [comment]
It works fine. A also tested by only adding the change to match only */msr to discard the microcode file and got no warning anymore. So both fixes are fine.

**Pull request merged**